components/renku_data_services/namespace/orm.py

```diff
@@ -74,21 +75,43 @@ class NamespaceORM(BaseORM):
     )
     user: Mapped[UserORM | None] = relationship(lazy="joined", init=False, repr=False, viewonly=True)
 
+    @hybridproperty
```

Comment:
Why use `hybridproperty` and not the standard `property` here? There does not seem to be documentation available on this decorator.

Reply:
Quite confusingly, there is documentation about `sqlachemy.ext.hybrid.hybrid_property` -> https://docs.sqlalchemy.org/en/20/orm/extensions/hybrid.html.
But using that decorator does not work, or requires the use of `inplace.expression`.

Reply:
yes, code completion error, sqlalchemy has both and this is the wrong one (this one doesn't have any documentation).

According to the docs the hybrid_property does not necessarily need an inplace, if only properties on the ORM are accessed in simple ways it can just work. But with relations that's not the case.

Hybrid properties first because I wanted to see how they work and also because it'd allow filtering on them in queries instead of having to calculate the values in python code.

This works though not exactly how I'd hoped. It creates queries like:
```
SELECT common.namespaces.id, common.namespaces.slug, common.namespaces.group_id, common.namespaces.user_id, groups_1.id AS id_1, groups_1.name, groups_1.created_by, groups_1.creation_date, groups_1.description, users_1.keycloak_id, users_1.first_name, users_1.last_name, users_1.email, users_1.secret_key, users_1.id AS id_2
FROM common.namespaces LEFT OUTER JOIN common.groups AS groups_1 ON groups_1.id = common.namespaces.group_id LEFT OUTER JOIN users.users AS users_1 ON users_1.keycloak_id = common.namespaces.user_id, common.groups
WHERE coalesce(common.groups.created_by, common.namespaces.user_id) = $1::VARCHAR ORDER BY common.namespaces.id
 LIMIT $2::INTEGER OFFSET $3::INTEGER
```

So it does allow filtering in the query directly (see the `WHERE coalesce(...)` which comes from the hybrid_property, but it does not show up in the SELECT. I expected something like `SELECT coalesce(...) as created_by` there as well.

I though it nice if we can filter all namespaces with a specific creator or all namespaces after a date efficiently at no real cost. Though right now this isn't used, so I can also swap it out for a regular property. Was more of a learning experience :)